proposals/NNNN-non-member-operator-overloading.md

```diff
@@ -0,0 +1,66 @@
+# Non-member Operator Overloading
+
+## Instructions
+
+* Proposal: [NNNN](NNNN-non-member-operator-overloading.md)
+* Author(s): [Chris Bieneman](https://github.com/llvm-beanz)
+* Sponsor: [Chris Bieneman](https://github.com/llvm-beanz)
+* Status: **Under Consideration**
+* Planned Version: 202x
+* Related Proposal(s): [0006 Reference Types](0006-reference types.md)
+
+## Introduction
+
+HLSL 202x introduced operator overloading for member operators of user defined
+data types. Only supporting member operators has some drawbacks, specifically
+defining binary operators where the LHS is a built-in type is impossible,
+additionally scoping operators to namespaces is also impossible.
+
+## Motivation
+
+Beyond common use cases where global operators are nice to have, users adopting
+HLSL 2021 have provided feedback that global operator overloading would be a
+significant benefit.
+
+HLSL 2021's introduction of short circuiting boolean operators users have been
+forced to translate their code that operates on HLSL vector types to use the
+new `select`, `and` and `or` intrinsics. We believe the new intrinsics are the
+right step for the language in removing ambiguity and clearly expressing code
+meaning. Global operator overloading provides an opportunity for the language to
+remain clear and unambiguous, while allowing users to define custom operators
+that allow them to not migrate legacy code.
+
+## Proposed solution
+
+This change requires fully adopting C++ rules for operator overload resolution
+and supporting defining global and namespace scoped operator overloads. The
+limitations currently in place on operator overloading (disallowing overloading
+operators that idiomatically return references) will remain in effect on
+non-member operators as well unless the restrictions are lifted as proposed in
+[0006 Reference Types](0006-reference types.md).
+
+Because this solution should not break existing code, it could also be enabled
+under HLSL 2021 as an extension.
+
+## Alternatives considered

```

Comment:
These are alternatives to the motivation at hand, but there are other benefits to non-member operators that these don’t represent an alternative to. I realize I’m being nitpicky, but I’m trying to imagine coming at this without the background we have

Reply:
That's a fair observation and I should note it. Since much of the motivation for prioritizing this feature is addressing an immediate developer need I felt having this context is important.